### ISSUE TITLE
Note trigger timing drift bug

### ISSUE THREAD

**Issue by gdreyband** [bsutherland/JuceOPLVSTi#64]
I'd like to pick up where we left off and it's been a while since I've been here so I'll mark a few issues that I'm hoping are minor but realistically probably aren't.

1) Timing drift - The timing of note triggers isn't super consistent. I have two kicks on a track that sometimes hit together but mostly don't. There seems to be some timing drift.

2) The attenuation sliders are too short which means intermediate values are skipped. I've attached my revised version of the PluginGui files which should correct it.

[JUCEOPLVSTi PluginGui files.zip](https://github.com/bsutherland/JuceOPLVSTi/files/1508504/JUCEOPLVSTi.PluginGui.files.zip)

**Comment by bsutherland** [comment]
Hi Gene,

Thanks for your contributions as always. I last worked on the code back in June, but it's in a half-completed state, so I didn't push it. I've incorporated most of your ideas and fixed the slider length, but it's still a work in progress, sorry about that.

About the timing drift, do you mean that you trigger them at the same time in your host, but the plugin plays them at different times? Thanks.

**Comment by bsutherland** [comment]
Just wanted to let you know, I updated all my tools and pushed the code.
I'll try to make some time to work on the plugin again before the end of the year.
If you'd like to contribute changes in the mean time (definitely appreciated!), please fork the repo and create a pull-request when you have it in a state which builds and is ready to merge.
Thanks!

**Comment by gdreyband** [comment]
Yes, sorry I wasn't clear. I was doubling the kicks since 2-OP can be kind of limited but sometimes they land together and other times they don't, even though they're triggered at the same time.

As a side note, I discovered this great tutorial a few months for making audio plugins: http://www.martin-finke.de/blog/tags/making_audio_plugins.html

The creator of the tutorial, Martin Finke created this repo: https://github.com/martinfinke/ReaX

Apparently ReaX makes it easier to hook up parameters to the GUI. I'm going to try to learn how to incorporate it into a JUCE project so it reduces the burden of creating the 4-OP OPL3 version since there's the huge headache of assigning all of the parameters 4x plus the added algorithms. I'll be trying out a demo project at some point just to see if I can get the hang of it. Just wanted to make you aware of it if you hadn't heard of it before.

**Comment by bsutherland** [comment]
Looks like a nice tutorial. Have fun working through it, hope you can get the hang of it.
